src/replication.c

```diff
@@ -1325,10 +1340,9 @@ void replconfCommand(client *c) {
                     addReplyErrorFormat(c,"Rdb channel sync is not allowed when repl-diskless-sync disabled on primary");
                     return;
                 }
-                c->flags |= CLIENT_RDB_CHANNEL_PSYNCCHAN;
-                c->slave_capa |= SLAVE_CAPA_PSYNC2;
+                c->flags |= CLIENT_REPL_MAIN_CHANNEL;
```

Comment:
Initially, I thought the issue was enabling PSYNC2 in case the replica asked for an end offset. Although I agree that we almost have `no-fullsync` or `psync-only` for free, I think that implementing such behavior will require more than simply adding REPLCONF options. With proper design and testing, I will be happy to add this feature later on. In the meantime, I'll  rename `no-fullsync` to `main-channel`.